File: server/src/main/java/org/hkurh/doky/entities/WorkflowEntity.java
package org.hkurh.doky.entities;

import javax.persistence.*;

@Entity
@Table(name = "workflow")
public class WorkflowEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "code", nullable = false, unique = true)
    private String code;

    @Column(name = "name")
    private String name;

    @Lob
    @Column(name = "description")
    private String description;

    @ManyToOne
    @JoinColumn(name = "document_type_id")
    private DocumentTypeEntity documentType;

    public DocumentTypeEntity getDocumentType() {
        return documentType;
    }

    public void setDocumentType(DocumentTypeEntity documentType) {
        this.documentType = documentType;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
}

File: server/src/main/java/org/hkurh/doky/workflow/WorkflowEntity.kt
package org.hkurh.doky.workflow

import jakarta.persistence.*
import org.hkurh.doky.documents.db.DocumentTypeEntity

@Entity
@Table(name = "workflow")
class WorkflowEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    var id: Long = -1

    @Column(name = "code", nullable = false, unique = true)
    var code: String? = null

    @Column(name = "name")
    var name: String? = null

    @Lob
    @Column(name = "description")
    var description: String? = null

    @ManyToOne
    @JoinColumn(name = "document_type_id")
    var documentType: DocumentTypeEntity? = null
}
